src/m365/planner/commands/bucket/bucket-get.spec.ts

```diff
@@ -269,24 +269,23 @@ describe(commands.BUCKET_GET, () => {
       return Promise.reject('Invalid Request');
     });
 
-    command.action(logger, {
-      options: {
-        name: validBucketName,
-        planTitle: validPlanTitle,
-        ownerGroupName: validOwnerGroupName
-      }
-    }, (err?: any) => {
-      try {
-        assert.strictEqual(JSON.stringify(err), JSON.stringify(new CommandError(`The specified group '${validOwnerGroupName}' does not exist.`)));
-        done();
-      }
-      catch (e) {
-        done(e);
-      }
-    });
+    try {
+      await command.action(logger, {
```

Comment:
Let's see if we can use `assert.throws` which I believe is made for this use-case, instead of using custom logic

Reply:
Hi @waldekmastykarz 

`assert.throws` doesn't seem to work with promises. I found out `assert.rejects` is the equivalent which does support promises.
Just tested it out, and seems to work as expected. The test of this review would look like this.

```typescript
  it('fails validation when no groups found', async () => {
    sinon.stub(request, 'get').callsFake((opts) => {
      if (opts.url === `https://graph.microsoft.com/v1.0/groups?$filter=displayName eq '${encodeURIComponent(validOwnerGroupName)}'`) {
        return Promise.resolve({ "value": [] });
      }

      return Promise.reject('Invalid Request');
    });

    await assert.rejects(command.action(logger, {
        options: {
          name: validBucketName,
          planTitle: validPlanTitle,
          ownerGroupName: validOwnerGroupName
        }
      }, new CommandError(`The specified group '${validOwnerGroupName}' does not exist.`);
  });
```

The test does fail when the message doesn't match. Is the example above all right to implement in all other tests?

Reply:
I've updated the `planner bucket` tests with their new version. Could you check if this is the right way forward please?

Reply:
You're right. Apparently, with async functions you need to use `assert.rejects` instead of `assert.throws`. 👍

Reply:
Yeah first I refactored the test to use async with `done`. However the test builds, when I run it, I do get an error message (can't remember the message anymore). After a quick Google I found out it was because of the `done`.
The done is used as callback for the test function, however when using `async` this is not needed anymore. If the `await command.action(...` throws an exception, the test fails.

Reply:
As long as it's just the test that's failing and not the whole runner, and the test is properly marked as failed, we could drop it to simplify our code.

Reply:
That was what I experienced. The test fails and the next one is executed. I'll check this behavior again just to be sure I don't replace all tests with code that doesn't work as it should. 😀

Reply:
Just rechecked. When an uncaught exception is thrown, the test fails and the next one is executed. This also without the `done`. So I think this is exactly what we need.
![image](https://user-images.githubusercontent.com/11723921/185460648-f36f5005-0d44-41b0-827b-fa6f7a770a6b.png)